smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -44,6 +44,62 @@ contract MixinTransfer is
   // This is calculated as `keyPrice * transferFeeBasisPoints / BASIS_POINTS_DEN`.
   uint public transferFeeBasisPoints;
 
+  /**
+  * @notice Allows the key owner to share their key (parent key) by
+  * transferring a portion of the remaining time to a new key (child key).
+  * @param _from The owner of the parent key
+  * @param _to The recipient of the shared key
+  * @param _tokenId the key to share
+  * @param _timeShared The amount of time shared
+  */
+  function shareKey(
+    address _from,
```

Comment:
Why accept _from?

By using onlyKeyOwnerOrApproved you are confirming the msg.sender has access to that tokenId, but then in the code below you pull the key indexed by _from.

Reply:
Well, first, we want to make sure that `_from` has a valid key before trying to share it, and second, with from we can look up the key in one step, vs using `_tokenId` to look up the owner, then looking up the key.

Reply:
if _from as approved to spend _tokenId (which is owned by someone else), this function proceeds and then _fromKey.tokenId != _tokenId

Reply:
I was thinking that even if this function is called by someone else who is approved, `_from` is still the address of the key owner, not msg.sender.

Reply:
We could remove this param, and just use `_tokenId` to look up the owner, and then get the key and check its validity using that...

Reply:
But we can't assume the `_from` address is the message.sender. In fact, as per the interface, `_from` is "The owner of the parent key"

Reply:
We can change this if you like, but it's the exact same pattern we use in `transferFrom`

Reply:
resolved in slack channel.